streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -232,15 +272,30 @@ public StreamsConfig(Map<?, ?> props) {
         props.put(StreamsConfig.REPLICATION_FACTOR_CONFIG, getInt(REPLICATION_FACTOR_CONFIG));
         props.put(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, getInt(NUM_STANDBY_REPLICAS_CONFIG));
         props.put(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG, StreamPartitionAssignor.class.getName());
-
         if (!getString(ZOOKEEPER_CONNECT_CONFIG).equals(""))
             props.put(StreamsConfig.ZOOKEEPER_CONNECT_CONFIG, getString(ZOOKEEPER_CONNECT_CONFIG));
 
         return props;
     }
 
     public Map<String, Object> getRestoreConsumerConfigs(String clientId) {
-        Map<String, Object> props = getBaseConsumerConfigs();
+        // filter out non-consumer configs
+        Map<String, ConfigKey> configKeys = ConsumerConfig.configKeys();
+        Map<String, Object> props = filter(configKeys, this.originals());
+
+        // disable auto commit ignoring any user overridden values,
+        // this is necessary for streams commit semantics
+        if (props.containsKey(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG)) {
+            log.warn("Ignoring user specified consumer config " + ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG
+                    + " and always turn off auto committing.");
+            props.remove(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG);
+        }
+
+        // enforce streams overrides if not specified by user
+        for (String keyName : CONSUMER_OVERRIDES.keySet()) {
+            if (!props.containsKey(keyName))
+                props.put(keyName, CONSUMER_OVERRIDES.get(keyName));
+        }
```

Comment:
It seems like the code here is the same as the code in `getConsumerConfigs`. Probably worth extracting into a private method.